marketplaceItems/drawApp/fingerDraw_app.js

```diff
@@ -0,0 +1,596 @@
+    var mouseEventsConnected = false;
+    function setUpDesktopMode() {
+        if (!mouseEventsConnected) {
+            mouseEventsConnected = true;
+            Controller.mousePressEvent.connect(mousePressed);
+            Controller.mouseMoveEvent.connect(mouseContinueLine);
+            Controller.mouseReleaseEvent.connect(mouseReleased);
+        }
+    }
+
+    /* CLOSE HMD MODE: Remove controller mapping */
+    function closeHMDMode() {
+        if (controllerMapping) {
+            controllerMapping.disable();
+        }
+        Messages.sendLocalMessage("Hifi-Hand-Disabler", "none");
+        if (animationHandlerID) {
+            animationHandlerID = MyAvatar.removeAnimationStateHandler(animationHandlerID);
+        }
+    }
+
+    /* CLOSE DESKTOP MODE: Stop listening for mouse presses */
+    function closeDesktopMode() {
+        if (mouseEventsConnected) {
+            mouseEventsConnected = false;
+            Controller.mousePressEvent.disconnect(mousePressed);
+            Controller.mouseMoveEvent.disconnect(mouseContinueLine);
+            Controller.mouseReleaseEvent.disconnect(mouseReleased);
+        }
+    }
+
+    /* WHEN USER DOMAIN CHANGES: Close app to remove paint sphere in hand when leaving the domain */
+    var WAIT_TO_CLEAN_UP = 2000;
+    function domainChanged() {
+        Script.setTimeout(function() {
+            cleanUp();
+        }, WAIT_TO_CLEAN_UP);
+    }
+
+    /* WHEN USER CHANGES DOMINANT HAND: Switch default hand to place paint sphere in */
+    var WAIT_TO_REOPEN_APP_MS = 500;
+    function handChanged() {
+        if (MyAvatar.getDominantHand() === dominantHand) {
+            return;
+        }
+        dominantHand = MyAvatar.getDominantHand();
+        dominantHandJoint = (dominantHand === "right") ? "RightHand" : "LeftHand";
+        if (paintSphere) {
+            onClicked();
+        }
+        if (distanceCheckInterval) {
+            Script.clearInterval(distanceCheckInterval);
+            distanceCheckInterval = null;
+        }
+        if (deletingInterval) {
+            Script.clearInterval(deletingInterval);
+            deletingInterval = null;
+        }
+        Script.setTimeout(function() {
+            onClicked();
+        }, WAIT_TO_REOPEN_APP_MS);
+    }
+
+    var tablet = Tablet.getTablet('com.highfidelity.interface.tablet.system');
+    var button = tablet.addButton({
+        text: 'DRAW',
+        icon: Script.resolvePath('assets/icons/draw-i.png'),
+        activeIcon: Script.resolvePath('assets/icons/draw-a.png')
+    });
+    var dominantHand = MyAvatar.getDominantHand();
+    var dominantHandJoint = (dominantHand === "right") ? "RightHand" : "LeftHand";
+    MyAvatar.dominantHandChanged.connect(handChanged);
+    registerControllerMapping();
+    HMD.displayModeChanged.connect(displayModeChange);
+    button.clicked.connect(onClicked);
+    Window.domainChanged.connect(domainChanged);
+    Script.scriptEnding.connect(appEnding);
```

Comment:
could you please add functionality that connects a signal to `tablet.tabletShownChanged()` that temporarily disables the draw app functionality when the tablet is open? we want users to be able to use the tablet when they have the app enabled and they open the tablet.